processor/src/chiplets/hasher/mod.rs

```diff
@@ -285,6 +426,12 @@ impl Hasher {
         let index_bit = *index & 1;
         let mut state = build_merge_state(&root, &sibling, index_bit);
 
+        // add the lookup for the hash initialization if this is the beginning.
+        let context = HasherLookupContext::Start;
+        if let Some(label) = get_selector_context_label(init_selectors, context) {
+            self.append_lookup(label, state, Felt::new(*index), context);
+        }
+
```

Comment:
Obviously there will be many times that this function will be called and there will be no selector match either here or below (with the `final_selectors`). I opted to do it this way since we need the state information which is only available inside this function & there were fewer changes this way. The alternative would be to refactor `verify_mp_leg`, but it seemed worthwhile to have a working approach first.

Reply:
Let's leave a bigger refactoring for the future.